### PR TITLE
Np 10520/configure analytic events nbp

### PR DESCRIPTION
**Pull request by LenoreChen**
Adds support for analytic events logging
* export `logAnalyticEvent` from `ApplicationController.js` for logging 
* add specific events such as user id and user email
* create trace ID in `AnalyticEventsAgent` which can be configured to tract unique attempts from user
* add user clicks to default logging to identify when wizard failed to save

### CODE REVIEW

File: src/foam/nanos/controller/ApplicationController.js [hunk 1]
```diff
@@ -1042,6 +1044,16 @@ foam.CLASS({
...
+    },
+    function logAnalyticEvent(evtName, evtTraceId, evtSessionId, evtExtra) {
+      this.__subContext__.analyticEventDAO.put(foam.nanos.analytics.AnalyticEvent.create(
+        {
+          name: evtName,
+          sessionId: evtSessionId,
+          traceId: evtTraceId,
+          extra: evtExtra
+        }
+      ));
```
Comment (sarthak-marwaha) [inline src/foam/nanos/controller/ApplicationController.js]:
```suggestion
      ), this);
```

Comment (sarthak-marwaha) [reply src/foam/nanos/controller/ApplicationController.js]: Actually, the applicationController should import 'analyticEventDAO' and require the AnalyticEvent class, this should just be 
`this.analyticEventDAO?.put(this.AnalyticEvent.create({....}))`

File: src/foam/u2/wizard/controllers/IncrementalWizardController.js [hunk 2]
```diff
@@ -17,6 +17,10 @@ foam.CLASS({
     'foam.u2.wizard.axiom.WizardAction'
   ],
 
+  imports: [
+    'handleEvent?'
```
Comment (sarthak-marwaha) [inline src/foam/u2/wizard/controllers/IncrementalWizardController.js]: Is this required id the parent class imports it as well?

Comment (LenoreChen) [reply src/foam/u2/wizard/controllers/IncrementalWizardController.js]: no need, removed

File: src/foam/nanos/controller/ApplicationController.js (hunk 1 above)
Comment (LenoreChen) [reply src/foam/nanos/controller/ApplicationController.js]: updated

**Review by sarthak-marwaha — APPROVED** [review]

**Pull request merged**